crates/data-plane-controller/src/stack.rs

```diff
@@ -115,13 +115,26 @@ pub struct AWSAssumeRole {
     pub external_id: String,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(untagged)]
+pub enum PrivateLink {
+    AWS(AWSPrivateLink),
+    Azure(AzurePrivateLink),
+}
```

Comment:
The alternative here is having two separate properties `aws_private_links` and `azure_private_links` and migrating existing `private_links` to `aws_private_link` with an alias. I don't have a strong feeling against the untagged enum here.

Reply:
I don't either, this seems perfectly workable